scripts/utils/options.ts

```diff
@@ -0,0 +1,245 @@
+}
+
+function shortFlag(key: OptionId, option: Option) {
+  const inverse = isBooleanOption(option) && option.inverse;
+  const defaultShortFlag = inverse ? key.substring(0, 1).toUpperCase() : key.substring(0, 1);
+  const short = option.shortFlag || defaultShortFlag;
+  if (short.length !== 1) {
+    throw new Error(
+      `Invalid shortFlag for ${key}: '${short}', needs to be a single character (e.g. 's')`
+    );
+  }
+  return short;
+}
+
+function longFlag(key: OptionId, option: Option) {
+  const inverse = isBooleanOption(option) && option.inverse;
+  return inverse ? `no-${kebabCase(key)}` : kebabCase(key);
+}
+
+function optionFlags(key: OptionId, option: Option) {
+  const base = `-${shortFlag(key, option)}, --${longFlag(key, option)}`;
+  if (isStringOption(option) || isStringArrayOption(option)) {
+    return `${base} <${key}>`;
+  }
+  return base;
+}
+
+export function getOptions<TOptions extends OptionSpecifier>(
+  command: Command,
+  options: TOptions,
+  argv: string[]
+): MaybeOptionValues<TOptions> {
+  Object.entries(options)
+    .reduce((acc, [key, option]) => {
+      const flags = optionFlags(key, option);
+
+      if (isBooleanOption(option)) return acc.option(flags, option.description, !!option.inverse);
+
+      const checkStringValue = (raw: string) => {
+        if (!option.values.includes(raw))
+          throw new Error(`Unexpected value '${raw}' for option '${key}'`);
+        return raw;
+      };
+
+      if (isStringOption(option))
+        return acc.option(flags, option.description, (raw) => checkStringValue(raw));
+
+      if (isStringArrayOption(option)) {
+        return acc.option(
+          flags,
+          option.description,
+          (raw, values) => [...values, checkStringValue(raw)],
+          []
+        );
+      }
+
+      throw new Error(`Unexpected option type '${key}'`);
+    }, command)
+    .parse(argv);
+
+  // Note the code above guarantees the types as they come in, so we cast here.
+  // Not sure there is an easier way to do this
+  return command.opts() as MaybeOptionValues<TOptions>;
+}
+
+export function areOptionsSatisfied<TOptions extends OptionSpecifier>(
+  options: TOptions,
+  values: MaybeOptionValues<TOptions>
+) {
+  return !Object.entries(options)
+    .filter(([, option]) => isStringOption(option) && option.required)
+    .find(([key]) => !values[key]);
+}
+
+export async function promptOptions<TOptions extends OptionSpecifier>(
+  options: TOptions,
+  values: MaybeOptionValues<TOptions>
+): Promise<OptionValues<TOptions>> {
+  const questions = Object.entries(options).map(([key, option]): PromptObject => {
+    if (!isBooleanOption(option)) {
+      const currentValue = values[key];
+      return {
+        type: isStringArrayOption(option) ? 'autocompleteMultiselect' : 'select',
+        message: option.description,
+        name: key,
+        // warn: ' ',
+        // pageSize: Object.keys(tasks).length + Object.keys(groups).length,
```

Comment:
Are these comments intentional? Would be awesome to have a pageSize option

Reply:
Nope, was just copying from somewhere else. We can add more things as we need them.